datanode/gateway/graphql/oracles_resolver.go

```diff
@@ -14,43 +14,45 @@ package gql
 
 import (
 	"context"
+	"errors"
 	"strconv"
 
 	v2 "code.vegaprotocol.io/vega/protos/data-node/api/v2"
+	types "code.vegaprotocol.io/vega/protos/vega"
 	vegapb "code.vegaprotocol.io/vega/protos/vega"
+	v1 "code.vegaprotocol.io/vega/protos/vega/data/v1"
 )
 
 type oracleSpecResolver VegaResolverRoot
 
 func (o *oracleSpecResolver) DataSourceSpec(ctx context.Context, obj *vegapb.OracleSpec) (*ExternalDataSourceSpec, error) {
-	if obj.ExternalDataSourceSpec == nil {
-		return nil, nil
+	dataSourceSpec := &DataSourceSpec{
+		Data: &DataSourceDefinition{},
 	}
 
-	dataSourceSpec := &DataSourceSpec{}
-	if obj.ExternalDataSourceSpec.Spec != nil {
-		if obj.ExternalDataSourceSpec.Spec.Data != nil {
-			updatedAt := strconv.FormatInt(obj.ExternalDataSourceSpec.Spec.UpdatedAt, 10)
-			dataSourceSpec.ID = obj.ExternalDataSourceSpec.Spec.Id
-			dataSourceSpec.CreatedAt = strconv.FormatInt(obj.ExternalDataSourceSpec.Spec.CreatedAt, 10)
-			dataSourceSpec.UpdatedAt = &updatedAt
-			dataSourceSpec.Status = DataSourceSpecStatus(obj.ExternalDataSourceSpec.String())
-
-			dataSourceSpec.Data = &DataSourceDefinition{
-				SourceType: obj.ExternalDataSourceSpec.Spec.Data.SourceType.(DataSourceKind),
+	if obj.ExternalDataSourceSpec != nil {
+		if obj.ExternalDataSourceSpec.Spec != nil {
+			if obj.ExternalDataSourceSpec.Spec.Data != nil {
+				dataSourceSpec = resolveDataSourceSpec(obj.ExternalDataSourceSpec.Spec)
 			}
 		}
 	}
```

Comment:
```suggestion
	if ed := obj.ExternalDataSourceSpec; ed != nil && ed.Spec != nil && ed.Spec.Data != nil {
		dataSourceSpec = resolveDataSourceSpec(ed.Spec.Data)
	}
```

Reply:
This has been addressed here https://github.com/vegaprotocol/vega/pull/6750/commits/439e167fbb5953adcc6b6d7a481ea70596b19512#diff-0cde61576df49d23ac0103aca13202a605d1c83edf31d12fdc3d5281c83005c9R32-R33
Thanks!